### PR TITLE
8316447: 8 sun/management/jmxremote tests ignore VM flags

### PR DESCRIPTION
**Pull request by lmesnik**
Updated test to use createTesJvm.
Removed internal timeout to not fail when Xcomp is used and also to get more info if the test times out.,

Tested by running tier1, hs-tier5 and executing test with various VM flags.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8316447](https://bugs.openjdk.org/browse/JDK-8316447): 8 sun/management/jmxremote tests ignore VM flags (**Sub-task** - P4)

### Reviewers
 * [Kevin Walls](https://openjdk.org/census#kevinw) (@kevinjwalls - Committer) ⚠️ Review applies to [b8ddc215](https://git.openjdk.org/jdk/pull/16048/files/b8ddc215bf558f8eab0658a4ab8d6e995ceb9a67)
 * [Serguei Spitsyn](https://openjdk.org/census#sspitsyn) (@sspitsyn - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/16048/head:pull/16048` \
`$ git checkout pull/16048`

Update a local copy of the PR: \
`$ git checkout pull/16048` \
`$ git pull https://git.openjdk.org/jdk.git pull/16048/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 16048`

View PR using the GUI difftool: \
`$ git pr show -t 16048`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/16048.diff">https://git.openjdk.org/jdk/pull/16048.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/16048#issuecomment-1748086516)

### CODE REVIEW

**Review by kevinjwalls — APPROVED** [review]
Looks good - 

test/jdk/sun/management/jmxremote/bootstrap/JMXAgentInterfaceBinding.java
        private boolean jmxAgentStarted = false;
...this line can be removed as well I think.
		
		
ProcessTools does have createTestJvm(List<String> command), so the frequent toArray seems unnecessary.
The method does toArray itself 8-) and is recent compared to some of the tests,
so maybe the tests haven't changed since ProcessTools createJavaProcessBuilder and createTestJvm added versions that takes a List.

**Comment by lmesnik** [comment]
Thank you for your review and feedback.
I removed 
private boolean jmxAgentStarted = false;
The changes invocations to createTestJvm are needed because it added tested flags only.

**Comment by kevinjwalls** [comment]
> Thank you for your review and feedback. I removed private boolean jmxAgentStarted = false; The changes invocations to createTestJvm are needed because it added tested flags only.

OK thanks.
I was observing that we want to call ProcessTools.createTestJvm(), as it calls createJavaProcessBuilder(Utils.prependTestJavaOpts(command)).

When we have the arguments/command in a List, there is a createTestJvm method that takes a List, but here we convert to an array and pass that.

**Comment by lmesnik** [comment]
Thanks, I simplified tests.

File: test/jdk/sun/management/jmxremote/bootstrap/CustomLauncherTest.java [hunk 1]
```diff
@@ -95,9 +95,7 @@ public static void main(String[] args) throws Exception {
             System.out.println("  PID           : " + serverPrc.pid());
             System.out.println("  shutdown port : " + port.get());
 
-            ProcessBuilder client = ProcessTools.createJavaProcessBuilder(
-                "-cp",
-                Utils.TEST_CLASS_PATH,
```
Comment (sspitsyn) [inline test/jdk/sun/management/jmxremote/bootstrap/CustomLauncherTest.java:100]: Just a question: Is the `-cp` options added by the `ProcessTools.createTestJvm()`?

**Review by sspitsyn — APPROVED** [review]
Looks good to me.
Thanks,
Serguei

File: test/jdk/sun/management/jmxremote/bootstrap/CustomLauncherTest.java (hunk 1 above)
Comment (lmesnik) [reply test/jdk/sun/management/jmxremote/bootstrap/CustomLauncherTest.java:100]: It is added even by ProcessTools.createJavaProcessBuilder(..) so it is not needed here.

**Pull request closed without merging**